static/app/components/feedback/useFetchFeedbackInfiniteListData.tsx

```diff
@@ -0,0 +1,305 @@
+
+const PER_PAGE = 10;
+
+type Unsubscribe = () => void;
+
+function startDateFromQueryView({start, statsPeriod}: QueryView): Date {
+  if (start) {
+    return new Date(start);
+  }
+  if (statsPeriod) {
+    const value = parseInt(statsPeriod, 10);
+    const unit = statsPeriod.endsWith('m')
+      ? 'min'
+      : statsPeriod.endsWith('h')
+      ? 'hour'
+      : statsPeriod.endsWith('d')
+      ? 'day'
+      : statsPeriod.endsWith('w')
+      ? 'week'
+      : undefined;
+    if (unit) {
+      const msdifference = formatDuration({
+        precision: 'ms',
+        style: 'count',
+        timespan: [value, unit],
+      });
+      return moment.utc().subtract(msdifference, 'ms').toDate();
+    }
+  }
+  throw new Error('Must pass either start or statsPeriod');
+}
+
+function endDateFromQueryView({end}: QueryView): Date {
+  if (end) {
+    return new Date(end);
+  }
+  return new Date();
+}
+
+class InfiniteListLoader {
+  private dispatch = new Dispatch();
+
+  private timestampToFeedback = new Map<number, HydratedFeedbackItem>();
+
+  public hasPrev: undefined | boolean = undefined;
+  public hasMore: undefined | boolean = undefined;
+  public totalHits: undefined | number = undefined;
+
+  private isFetching = false;
+
+  constructor(
+    private api: Client,
+    private organization: Organization,
+    private queryView: QueryView,
+    private initialDate: Date
+  ) {
+    if (!queryView.start && !queryView.statsPeriod) {
+      return;
+    }
+    this.fetch({
+      start: startDateFromQueryView(queryView),
+      end: this.initialDate,
+      sort: '-timestamp',
+      perPage: PER_PAGE,
+    }).then(results => {
+      this.totalHits = results.hits;
+      this.didFetchNext(results);
+    });
+  }
+
+  get feedbacks() {
+    const feedbacks: HydratedFeedbackItem[] = [];
+    const initialDateTime = this.initialDate.getTime();
+    for (const [timestamp, feedback] of this.timestampToFeedback.entries()) {
+      if (timestamp < initialDateTime) {
+        feedbacks.push(feedback);
+      }
+    }
+    return feedbacks;
+  }
+
+  onChange(handler: () => void): Unsubscribe {
+    this.dispatch.addEventListener('change', handler);
+    return () => this.dispatch.removeEventListener('change', handler);
+  }
+
+  private get minDatetime() {
+    return (
+      new Date(Math.min(...Array.from(this.timestampToFeedback.keys()))) ?? new Date()
+    );
+  }
+
+  private get maxDatetime() {
+    return (
+      new Date(Math.max(...Array.from(this.timestampToFeedback.keys()))) ?? new Date()
+    );
+  }
+
+  public resetInitialTimestamp() {
+    this.initialDate = new Date(this.maxDatetime);
+  }
+
+  private async fetch({
+    end,
+    perPage,
+    sort,
+    start,
+  }: {
+    end: Date;
+    perPage: number;
+    sort: 'timestamp' | '-timestamp';
+    start: Date;
+  }) {
+    if (this.isFetching) {
+      return {
+        feedbacks: [],
+        hasNextPage: undefined,
+        hits: 0,
+      };
```

Comment:
Is there no UI for loading state?

Reply:
not yet

but we do have `isFetching` inside the loader class which we can expose for that.